### DISCUSSION TITLE
如何能获取路由器的ip地址作为upstream

### DISCUSSION THREAD

**Discussion by XuuKoo** [IrineSistiana/mosdns#224] [Q&A]
想境内域名直接使用运营商提供的dns，海外域名才走公共dns解析。
我的意思是除了手动写路由器内网地址，mosdns能否自动找到路由器ip，因为总不能换个网络就得改一次配置文件。

**Comment by rampageX** [comment]
https://github.com/IrineSistiana/mosdns/discussions/213

**Reply by XuuKoo** [reply]
> #213

思路明白，就是不知道windows下如何实现。。。

**Comment by rampageX** [comment]
getdns.cmd:
```
@echo off
echo exit | nslookup | for /f "tokens=2" %%a in ('findstr Address') do @echo %%a
```

mosdns.yaml:
```
...
- addr: ${{getdns.cmd}}
...
```

**Reply by XuuKoo** [reply]
谢谢。

但是我clash开了tun，代理mosdns的远程dns解析，结果返回的是保留地址。。。

clash 配置：

> dns:
>   enable: true
>   listen: 0.0.0.0:53
>   ipv6: false
>   enhanced-mode: redir-host
>   nameserver:
>     - tcp://127.0.0.1:1053
> 
> tun:
>   enable: true
>   stack: system
>   dns-hijack:
>     - 198.18.0.2:53
>   auto-route: true
>   auto-detect-interface: true

mosdns 配置：

> plugin:
>   - tag: mian_server
>     type: server
>     args:
>       server:
>         - addr: 127.0.0.1:1053
>           protocol: udp
>         - addr: 127.0.0.1:1053
>           protocol: tcp
>           idle_timeout: 120

**Reply by IrineSistiana** [reply]
既然都用 mosdns 了，何必还用运营商的 dns 呢。公用dot，doh 它们不香么。

**Reply by XuuKoo** [reply]
> 既然都用 mosdns 了，何必还用运营商的 dns 呢。公用dot，doh 它们不香么。

国内网站用运营商dns解析并没有什么坏处啊，主要是延迟更低。 海外网站的解析走代理。
